controllers/managedocs_controller.go

```diff
@@ -255,7 +263,22 @@ func (r *ManagedOCSReconciler) SetupWithManager(mgr ctrl.Manager, ctrlOptions *c
 		},
 	)
 
-	return ctrl.NewControllerManagedBy(mgr).
+	var options client.Options
+	var err error
+
+	options.Scheme = runtime.NewScheme()
+	utilruntime.Must(apiextenstionsv1.AddToScheme(options.Scheme))
+
+	egreeNetworkPolicyCRD := &unstructured.Unstructured{}
+	egreeNetworkPolicyCRD.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "apiextensions.k8s.io",
+		Kind:    "CustomResourceDefinition",
+		Version: "v1",
+	})
```

Comment:
There should be a very specific reason to use `Unstructured` instead of structured structs. Can you please explain why you choose to use it?

Reply:
Using a structured object and a restricted client gives me an error "the cache is not started, can not read objects", and if I use a structured object with an unrestricted client it returns the resource as NotFound.

Reply:
In the case of 4.10 cluster with SDN, it will return that the resource is notFound and hence the Network type will be changed to OVN and it will try to deploy egressfirewall instead of egressnetworkpolicy

Reply:
done